app/lib/account/consent_backend.dart

```diff
@@ -0,0 +1,176 @@
+        final userKey = _db.emptyKey.append(User, id: userId);
+        final query = _db.query<Consent>(ancestorKey: userKey)
+          ..filter('dedupId =', dedupId);
+        final list = await query.run().toList();
+        if (list.isNotEmpty) {
+          final old = list.first;
+          if (old.isExpired()) {
+            // expired entries should be deleted
+            await _delete(old);
+          } else if (old.shouldNotify()) {
+            // non-expired entries just re-send the notification
+            return await _sendNotification(old);
+          } else {
+            return InviteStatus(false, old.nextNotification);
+          }
+        }
+        // Create a new entry.
+        final consent = Consent.init(
+          parentKey: userKey,
+          type: type,
+          args: args,
+          fromUserId: authenticatedUser.userId,
+          descriptionText: descriptionText,
+          descriptionHtml: descriptionHtml,
+        );
+        await _db.commit(inserts: [consent]);
+        return await _sendNotification(consent);
+      });
+    });
+  }
+
+  Future<InviteStatus> _sendNotification(Consent consent) async {
+    final invitedEmail = await accountBackend.getEmailOfUserId(consent.userId);
+    await emailSender.sendMessage(createInviteEmail(
+      activeAccountEmail: authenticatedUser.email,
+      invitedEmail: invitedEmail,
+      message: consent.descriptionText,
+      consentUrl: consentUrl(consent.consentId),
+    ));
+    return await _db.withTransaction((tx) async {
+      final c = (await tx.lookup<Consent>([consent.key])).single;
+      c.notificationCount++;
+      c.lastNotified = DateTime.now().toUtc();
+      tx.queueMutations(inserts: [c]);
+      await tx.commit();
+      return InviteStatus(true, c.nextNotification);
+    }) as InviteStatus;
+  }
+
+  /// Removes obsolete/expired [Consent] entries from Datastore.
+  Future deleteObsoleteConsents() async {
+    final query = _db.query<Consent>()
+      ..filter('expires <', DateTime.now().toUtc());
+    await for (var entry in query.run()) {
+      try {
+        await _delete(entry);
+      } catch (e) {
+        _logger.info(
+            'Delete failed: ${entry.userId} ${entry.type} ${entry.args}', e);
+      }
+    }
+  }
+
+  Future _accept(Consent consent) async {
+    final action = _actions[consent.type];
+    await action?.onAccept(consent.userId, consent.args);
+    await _db.commit(deletes: [consent.key]);
+  }
+
+  Future _delete(Consent consent) async {
+    final action = _actions[consent.type];
+    await action?.onDelete(consent.userId, consent.args);
+    await _db.commit(deletes: [consent.key]);
+  }
+}
+
+class InviteStatus {
+  final bool notificationSent;
+  final DateTime nextNotification;
+
+  InviteStatus(this.notificationSent, this.nextNotification);
+}
+
+/// Callback that will be called on consent actions.
+abstract class ConsentAction {
+  /// Callback on accepting the consent.
```

Comment:
I suggest we add:
 * `renderEmailSubject(List<String> args)`
 * `renderEmailContents(String activeAccountEmail, String invitedEmail, String consentUrl, List<String> args)`
 * `renderConsentDescription(String activeAccountEmail, String invitedEmail, String consentUrl, List<String> args)`

This way the templates for these can exist in the locations where we define the consent `type`.